erpnext/stock/report/stock_balance/stock_balance.py

```diff
@@ -143,15 +141,24 @@ def get_item_warehouse_map(self):
 		item_warehouse_map = {}
 		self.opening_vouchers = self.get_opening_vouchers()
 
-		for entry in self.sle_entries:
-			group_by_key = self.get_group_by_key(entry)
-			if group_by_key not in item_warehouse_map:
-				self.initialize_data(item_warehouse_map, group_by_key, entry)
+		with frappe.db.unbuffered_cursor():
+			if self.filters.get("show_stock_ageing_data"):
+				self.sle_entries = self.sle_query.run(as_dict=True)
```

Comment:
Don't do this inside unbuffered cursor, it will be slower than doing it outside. 


Rest seems fine.

Reply:
Fixed